### PR TITLE
Added abstract and author Request

### PR DESCRIPTION
**Pull request by GabrielMassana**

### CODE REVIEW

File: Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{
+    CCPAuthorRequest *request = [self requestForAPI];
+
+    NSMutableString *authorQuery = [[NSMutableString alloc] init];
+    
+    NSArray *terms = [searchTerm componentsSeparatedByString:@" "];
+    
+    for (NSInteger index = 0; index < terms.count; index++)
+    {
+        NSString *term = @"";
```
Comment (wibosco) [inline Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m]: Should this be set to nil?

File: Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        
+        if (index == 0)
+        {
+            term = [NSString stringWithFormat:@"author:%@", terms[index]];
+        }
+        else
+        {
+            term = [NSString stringWithFormat:@" author:%@", terms[index]];
```
Comment (wibosco) [inline Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m:33]: I don't follow this - will this allow you to search for multiple authors?

File: Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+//  Cocoapods
+//
+//  Created by Gabriel Massana on 7/2/16.
+//  Copyright © 2016 Gabriel Massana. All rights reserved.
+//
+
+#import "CCPAuthorRequest.h"
+
+@implementation CCPAuthorRequest
+
```
Comment (wibosco) [inline Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m:12]: Should we have pragma marks?

File: Cocoapods/Networking/Requests/Abstract/CCPRequest.m [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+@implementation CCPRequest
+
++ (instancetype)requestForAPI
+{
+    CCPRequest *request = [self request];
+    
+    return request;
+}
+
+- (void)updateRequestWithEndpoint:(NSString *)endpoint
```
Comment (wibosco) [inline Cocoapods/Networking/Requests/Abstract/CCPRequest.m:20]: What does this method do? is that you will have an existing request and then change it?

File: Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m (hunk 2 above)
Comment (GabrielMassana) [inline Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m:33]: It is more about name and username. 
If you search your name in the website, they are producing this URL: https://cocoapods.org/?q=william%20boles

For the API is something similar but with author: before every term:
http://search.cocoapods.org/api/v1/pods.flat.hash.json?query=author:william%20author:boles

File: Cocoapods/Networking/Requests/Abstract/CCPRequest.m (hunk 4 above)
Comment (GabrielMassana) [inline Cocoapods/Networking/Requests/Abstract/CCPRequest.m:20]: This method is doing almost the same as our - (void)setEndpoint:(NSString *)endpoint;
But I changed the name to not be set.
The function of the method is to merge the constant part of the API URL with the enpoint of this request.

Please, tell me if you can imagine a better name, I'll be happy to change it.

File: Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m (hunk 3 above)
Comment (GabrielMassana) [inline Cocoapods/Networking/Requests/Author/CCPAuthorRequest.m:12]: added

**Pull request merged**